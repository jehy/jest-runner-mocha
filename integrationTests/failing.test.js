'use strict';

const semver = require('semver');
const runJest = require('./runJest');

const normalize = res => res.replace(' [ERR_ASSERTION]', '');

it('Works when it has failing tests', () => {
  return expect(runJest('failing').then(normalize)).resolves.toMatchSnapshot();
});

const runV8Test = !semver.satisfies(process.version, '>= 10.12.0') ? it : xit;

runV8Test('Should fail v8 coverage on old node versions', () => {
  return expect(
    runJest('failing', ['--coverage', '--coverageProvider', 'v8']),
  ).resolves.toMatchSnapshot();
});
