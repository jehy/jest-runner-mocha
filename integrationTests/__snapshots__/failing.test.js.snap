// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should fail v8 coverage on old node versions 1`] = `
"FAIL integrationTests/__fixtures__/failing/__mocha__/__tests__/file.test.js
● Test suite failed to run
Node version v8.17.0 does not have coverage information!
Please use node >= 10.12.0 or babel coverage.
15 | async function collectV8Coverage() {
16 | if (!semver.satisfies(process.version, '>= 10.12.0')) {
> 17 | throw new Error(\`Node version \${process.version} does not have coverage information!
| ^
18 | Please use node >= 10.12.0 or babel coverage.\`);
19 | }
20 | v8CoverageInstrumenter = new CoverageInstrumenter();
at mocked-stack-trace
Test Suites: 1 failed, 1 total
Tests: 0 total
Snapshots: 0 total
Time:
Ran all test suites.

----------|---------|----------|---------|---------|-------------------
File | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files | 0 | 0 | 0 | 0 |
----------|---------|----------|---------|---------|-------------------
"
`;

exports[`Works when it has failing tests 1`] = `
"FAIL integrationTests/__fixtures__/failing/__mocha__/__tests__/file.test.js
My tests
✕ This test fails
My tests This test fails
AssertionError: 1 == 2
+ expected - actual
-1
+2
at mocked-stack-trace
Test Suites: 1 failed, 1 total
Tests: 1 failed, 1 total
Snapshots: 0 total
Time:
Ran all test suites.

"
`;
